### PR TITLE
Add basic tests and custom configuration for VS Code to execute them

### PR DESCRIPTION
**Pull request by StanislawSwierc**
I want to make some changes to the core package in the future. This tests adds protection from potential regressions.

The log entries used in this tests are coming from the file used in `examples/drain_bigfile_demo.py`

I decided to add a settings file as it wasn't immediately obvious to me what args I need for unittest to match the custom convention used in the project.

### CODE REVIEW

File: tests/drain_test.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
+import unittest
+from drain3.drain import Drain
+
+
+class DrainTest(unittest.TestCase):
+    def test_add_log_message(self):
+        model = Drain()
```
Comment (barakreif) [inline tests/drain_test.py:7]: without explicit parameters you can't really guarantee stable results

**Comment by barakreif** [comment]
adding both the settings and ignore prevents any future change and imposes you personal settings on the entire repo. 

also running a linter such as black could improve readability

File: .vscode/settings.json [hunk 2]
```diff
@@ -0,0 +1,19 @@
+{
```
Comment (davidohana) [inline .vscode/settings.json:1]: Please don't commit files with personal settings

File: tests/drain_test.py (hunk 1 above)
Comment (davidohana) [reply tests/drain_test.py:7]: I agree with @barakreif that ctor arguments shall be specified. explicitely.

**Review by davidohana — CHANGES_REQUESTED** [review]
Thanks for the PR! A good start for adding unit tests to Drain3.

File: .vscode/settings.json (hunk 2 above)
Comment (StanislawSwierc) [reply .vscode/settings.json:1]: These settings were inferred form the project. I would consider them project settings.

Here is example why I think it is important to save project settings files. This project diverged from the default convention of using `test*.py` pattern for tests files. It requires custom patten to be passed. I had to look it up in `unittest` documentation. I want to make it easier for others. One option would be to extend the `CONTRIBUTING.md` file and include instructions on how to configure our environment. Another is to save project settings files. I chose the latter.

```
>python -m unittest discover

----------------------------------------------------------------------
Ran 0 tests in 0.000s

OK
```
```
>python -m unittest discover --pattern "*_test.py" -s tests
...
----------------------------------------------------------------------
Ran 3 tests in 0.001s

OK
```

File: tests/drain_test.py (hunk 1 above)
Comment (StanislawSwierc) [reply tests/drain_test.py:7]: I consider default parameter part of the contract of the class. By using default constructor in tests I can rely on them to detect regressions which violate this contract.

For example. In one of my future contributions I plan to add a new parameter which will cap the max memory consumed by this model. If I specified all the parameters in tests I could miss the fact that default behavior changes and learn about it only when this change makes to production. 

I'm happy to add a second unit tests with all the parameters set.

File: .vscode/settings.json (hunk 2 above)
Comment (davidohana) [reply .vscode/settings.json:1]: I prefer to change the convention to the default one. You can just rename masking_test.py to test_masking.py

File: tests/drain_test.py (hunk 1 above)
Comment (davidohana) [reply tests/drain_test.py:7]: OK, so please add a second file

Comment (StanislawSwierc) [reply tests/drain_test.py:7]: done :)

File: .vscode/settings.json [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+        "--pattern",
+        "*_test.py"
+    ],
+    "python.testing.pytestEnabled": false,
+    "python.testing.nosetestsEnabled": false,
+    "python.testing.unittestEnabled": true,
+    "python.linting.pylintEnabled": false,
+    "python.linting.flake8Enabled": true,
+    "python.linting.enabled": true,
+    "python.formatting.provider": "black"
```
Comment (StanislawSwierc) [inline .vscode/settings.json:18]: >also running a linter such as black could improve readability

I totally agree. That's why I selected `black` as the formatting provider in the project settings file.

**Review by StanislawSwierc — COMMENTED** [review]
Thanks for the discussion regarding project settings file. I added it to help other execute tests I added. If you think that it's too much I can remove it.

File: .vscode/settings.json (hunk 2 above)
Comment (StanislawSwierc) [reply .vscode/settings.json]: Thanks! I've just renamed masking_test and removed settings file.

**Comment by StanislawSwierc** [comment]
@davidohana, please have a look at this PR when you have time. I think I addressed all the comments.

**Review by davidohana — APPROVED** [review]

**Pull request merged**